src/main/java/com/traveltime/sdk/TravelTimeSDK.java

```diff
@@ -118,6 +129,26 @@
         }
     }
 
+    private <T> void completeProtoFuture(
+        CompletableFuture<Either<TravelTimeError, T>> future,
+        ProtoRequest<T> protoRequest,
```

Comment:
Is there a way to structure it so you don't need the request to parse the response? Could `parseByteResponse` be static or something?

Reply:
I don't understand what exactly you want. I receive bytes after sending request, so I need to deserialize these bytes to object

Reply:
completeProtoFuture itself is static but I don't see the reason to mark it static when this method is private

Reply:
Never mind; back in Scala and in the protobuf-generated Java methods, `parseBytes` is a static method which doesn't require the request. 

I didn't understand at first why your parseByteResponse needs it, but you've made `ProtoRequest<T>` a generic and carry around the info which response is expected there so, I guess I understand now.